src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,126 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.Book;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.util.ConnectionCreator;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CANNOT_CREATE_BOOK_MSG = "Cannot create the book: ";
+    private static final String CANNOT_FIND_OBJECT_MSG = "Cannot find the needed object by id: ";
+    private static final String CANNOT_UPDATE_ROW_MSG = "Cannot update the row by ID: ";
+    private static final String FAILED_TO_DELETE_ROW = "Cannot delete the element. ID: ";
+    private static final String FAILED_TO_FIND_ROWS_MSG = "Failed to find all rows";
+    private static final String ID_COLUMN_NAME = "id";
+    private static final String NO_ROWS_INSERTED_MSG = "No rows inserted";
+    private static final String NO_NEEDED_ROW_MSG = "The required row is not present";
+    private static final String PRICE_COLUMN_NAME = "price";
+    private static final String SQL_SELECT_STATEMENT = "SELECT * FROM books WHERE id = ?";
+    private static final String SQL_INSERT_STATEMENT = "INSERT INTO books (title, price) "
+            + "VALUES (?, ?)";
+    private static final String SQL_DELETE_STATEMENT = "DELETE FROM books WHERE id = ?";
+    private static final String SQL_UPDATE_STATEMENT = "UPDATE books SET title = ?,"
+            + " price = ? WHERE id = ?";
+    private static final String SQL_SELECT_ALL_STATEMENT = "SELECT * FROM books";
+    private static final String TITLE_COLUMN_NAME = "title";
+    private static ConnectionCreator connectionCreator = new ConnectionCreator();

```

Comment:
Why is it a static member? If you decided to use a service with instance method instead of utility class with static method, then don't forget about SOLID (Dependency Inversion Principle), and why is it bad to assign dependencies in a class itself. (You can use an injector here, or inject it via constructor injection, but not like this).